hypershift-operator/metrics.go

```diff
@@ -485,6 +530,71 @@ func (m *hypershiftMetrics) observeNodePools(ctx context.Context, nodePools *hyp
 	return nil
 }
 
+func getOperatorImage(client crclient.Client) error {
+	ctx := context.TODO()
+	hypershiftNamespace := os.Getenv("MY_NAMESPACE")
+	hypershiftPodName := os.Getenv("MY_NAME")
+
+	hypershiftPod := &corev1.Pod{ObjectMeta: metav1.ObjectMeta{Name: hypershiftPodName, Namespace: hypershiftNamespace}}
+	if err := client.Get(ctx, crclient.ObjectKeyFromObject(hypershiftPod), hypershiftPod); err != nil {
+		if errors.IsNotFound(err) {
+			hypershiftDeployment := &appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Namespace: hypershiftNamespace, Name: assets.HypershiftOperatorName}}
```

Comment:
why is this fallback needed?

Reply:
If thereis no pod (for whatever reason), the metric will ask for the deployment, at least to have something in the metrics. Also, If the ImageID is not present in metrics, is a signal that the pod is not running and could help to identify a potential issue

Reply:
you're totally right, fixed already. thanks